Bug: Profile Page "Something went wrong..."

### What happened?

When I try clicking on the profile page in the top right where it displays your username and avatar, it brings me to an error page saying that "Something went wrong...".
![5JduBK](https://user-images.githubusercontent.com/78281961/213591050-3aae2fe6-703c-44fd-bbaa-201aad69ecf1.png)
I don't know what is wrong, this used to be working but suddenly just started producing this error.


### Version

latest (ghcr.io/diced/zipline:latest)

### What browser(s) are you seeing the problem on?

Firefox, Chromium-based (Chrome, Edge, Brave, Opera, mobile chrome/chromium based, etc), Safari, Firefox Mobile, Safari Mobile

### Zipline Logs

_No response_

### Browser Logs

```shell
framework-1aa899be9db520d4.js:9 Error: Menu component was not found in the tree
    at r (85516-3dcecd829fb618b5.js:1:21344)
    at 54915-a3ca6a0762003aa1.js:1:38032
    at ak (framework-1aa899be9db520d4.js:9:60918)
    at ui (framework-1aa899be9db520d4.js:9:71229)
    at i (framework-1aa899be9db520d4.js:9:121561)
    at oD (framework-1aa899be9db520d4.js:9:99115)
    at framework-1aa899be9db520d4.js:9:98982
    at oO (framework-1aa899be9db520d4.js:9:98989)
    at oE (framework-1aa899be9db520d4.js:9:95741)
    at ox (framework-1aa899be9db520d4.js:9:94296)
a9 @ framework-1aa899be9db520d4.js:9
main-e5277298906f66b4.js:1 Error: Menu component was not found in the tree
    at r (85516-3dcecd829fb618b5.js:1:21344)
    at 54915-a3ca6a0762003aa1.js:1:38032
    at ak (framework-1aa899be9db520d4.js:9:60918)
    at ui (framework-1aa899be9db520d4.js:9:71229)
    at i (framework-1aa899be9db520d4.js:9:121561)
    at oD (framework-1aa899be9db520d4.js:9:99115)
    at framework-1aa899be9db520d4.js:9:98982
    at oO (framework-1aa899be9db520d4.js:9:98989)
    at oE (framework-1aa899be9db520d4.js:9:95741)
    at ox (framework-1aa899be9db520d4.js:9:94296)
$ @ main-e5277298906f66b4.js:1
main-e5277298906f66b4.js:1 A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
$ @ main-e5277298906f66b4.js:1
framework-1aa899be9db520d4.js:9 Error: Menu component was not found in the tree
    at r (85516-3dcecd829fb618b5.js:1:21344)
    at 54915-a3ca6a0762003aa1.js:1:32439
    at ak (framework-1aa899be9db520d4.js:9:60918)
    at ui (framework-1aa899be9db520d4.js:9:71229)
    at i (framework-1aa899be9db520d4.js:9:121561)
    at oD (framework-1aa899be9db520d4.js:9:99115)
    at framework-1aa899be9db520d4.js:9:98982
    at oO (framework-1aa899be9db520d4.js:9:98989)
    at oE (framework-1aa899be9db520d4.js:9:95741)
    at ox (framework-1aa899be9db520d4.js:9:94296)
a9 @ framework-1aa899be9db520d4.js:9
main-e5277298906f66b4.js:1 Error: Menu component was not found in the tree
    at r (85516-3dcecd829fb618b5.js:1:21344)
    at 54915-a3ca6a0762003aa1.js:1:32439
    at ak (framework-1aa899be9db520d4.js:9:60918)
    at ui (framework-1aa899be9db520d4.js:9:71229)
    at i (framework-1aa899be9db520d4.js:9:121561)
    at oD (framework-1aa899be9db520d4.js:9:99115)
    at framework-1aa899be9db520d4.js:9:98982
    at oO (framework-1aa899be9db520d4.js:9:98989)
    at oE (framework-1aa899be9db520d4.js:9:95741)
    at ox (framework-1aa899be9db520d4.js:9:94296)
$ @ main-e5277298906f66b4.js:1
main-e5277298906f66b4.js:1 A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
$ @ main-e5277298906f66b4.js:1
main-e5277298906f66b4.js:1 Error rendering page:  Error: Cancel rendering route
    at c (main-e5277298906f66b4.js:1:12315)
    at main-e5277298906f66b4.js:1:12267
    at new Promise (<anonymous>)
    at ea (main-e5277298906f66b4.js:1:12244)
    at main-e5277298906f66b4.js:1:10856
```


### Additional Info

_No response_

--- a/src/components/Layout.tsx
+++ b/src/components/Layout.tsx
@@ -4,7 +4,6 @@ import {
   Box,
   Burger,
   Button,
-  Group,
   Header,
   Image,
   MediaQuery,
@@ -12,14 +11,11 @@ import {
   Navbar,
   NavLink,
   Paper,
-  Popover,
   ScrollArea,
   Select,
-  Stack,
   Text,
   Title,
   Tooltip,
-  UnstyledButton,
   useMantineTheme,
 } from '@mantine/core';
 import { useClipboard } from '@mantine/hooks';
@@ -55,61 +51,6 @@ import {
   UserIcon,
 } from './icons';
 import { friendlyThemeName, themes } from './Theming';
-
-function MenuItemLink(props) {
-  return (
-    <Link href={props.href} passHref legacyBehavior>
-      <MenuItem {...props} />
-    </Link>
-  );
-}
-
-function MenuItem(props) {
-  return (
-    <UnstyledButton
-      sx={(theme) => ({
-        display: 'block',
-        width: '100%',
-        padding: 5,
-        borderRadius: theme.radius.sm,
-        color: props.color
-          ? theme.fn.themeColor(props.color, theme.colorScheme === 'dark' ? 5 : 7)
-          : theme.colorScheme === 'dark'
-          ? theme.colors.dark[0]
-          : theme.black,
-        '&:hover': !props.noClick
-          ? {
-              backgroundColor: props.color
-                ? theme.fn.rgba(
-                    theme.fn.themeColor(props.color, theme.colorScheme === 'dark' ? 9 : 0),
-                    theme.colorScheme === 'dark' ? 0.2 : 1
-                  )
-                : theme.colorScheme === 'dark'
-                ? theme.fn.rgba(theme.colors.dark[3], 0.35)
-                : theme.colors.gray[0],
-            }
-          : null,
-      })}
-      {...props}
-    >
-      <Group noWrap>
-        <Box
-          sx={(theme) => ({
-            marginRight: theme.spacing.xs / 4,
-            paddingLeft: theme.spacing.xs / 2,
-
-            '& *': {
-              display: 'block',
-            },
-          })}
-        >
-          {props.icon}
-        </Box>
-        <Text size='sm'>{props.children}</Text>
-      </Group>
-    </UnstyledButton>
-  );
-}
 
 export type NavbarItems = {
   icon: React.ReactNode;
@@ -198,7 +139,6 @@ export default function Layout({ children, props }) {
   const [systemTheme, setSystemTheme] = useState(user.systemTheme ?? 'system');
   const version = useVersion();
   const [opened, setOpened] = useState(false); // navigation open
-  const [open, setOpen] = useState(false); // manage acc dropdown
 
   const avatar = user?.avatar ?? null;
   const router = useRouter();
@@ -375,11 +315,10 @@ export default function Layout({ children, props }) {
             </MediaQuery>
             <Title ml='sm'>{title}</Title>
             <Box sx={{ marginLeft: 'auto', marginRight: 0 }}>
-              <Popover position='bottom-end' opened={open} onClose={() => setOpen(false)}>
-                <Popover.Target>
+              <Menu>
+                <Menu.Target>
                   <Button
                     leftIcon={avatar ? <Image src={avatar} height={32} radius='md' /> : <SettingsIcon />}
-                    onClick={() => setOpen((o) => !o)}
                     sx={(t) => ({
                       backgroundColor: 'inherit',
                       '&:hover': {
@@ -392,84 +331,73 @@ export default function Layout({ children, props }) {
                   >
                     {user.username}
                   </Button>
-                </Popover.Target>
-
-                <Popover.Dropdown p={4} mr='md' sx={{ minWidth: '200px' }}>
-                  <Stack spacing={2}>
-                    <Menu>
-                      <Menu.Label>
-                        {user.username} ({user.id}){' '}
-                        {user.administrator && user.username !== 'administrator' ? '(Administrator)' : ''}
-                      </Menu.Label>
-                      <MenuItemLink icon={<SettingsIcon />} href='/dashboard/manage'>
-                        Manage Account
-                      </MenuItemLink>
-                      <MenuItem
-                        icon={<CopyIcon />}
-                        onClick={() => {
-                          setOpen(false);
-                          openCopyToken();
-                        }}
-                      >
-                        Copy Token
-                      </MenuItem>
-                      <MenuItem
-                        icon={<DeleteIcon />}
-                        onClick={() => {
-                          setOpen(false);
-                          openResetToken();
-                        }}
-                        color='red'
-                      >
-                        Reset Token
-                      </MenuItem>
-                      <MenuItemLink icon={<LogoutIcon />} href='/auth/logout' color='red'>
-                        Logout
-                      </MenuItemLink>
+                </Menu.Target>
+                <Menu.Dropdown>
+                  <Menu.Label>
+                    {user.username} ({user.id}){' '}
+                    {user.administrator && user.username !== 'administrator' ? '(Administrator)' : ''}
+                  </Menu.Label>
+                  <Menu.Item component={Link} icon={<SettingsIcon />} href='/dashboard/manage'>
+                    Manage Account
+                  </Menu.Item>
+                  <Menu.Item
+                    icon={<CopyIcon />}
+                    onClick={() => {
+                      openCopyToken();
+                    }}
+                  >
+                    Copy Token
+                  </Menu.Item>
+                  <Menu.Item
+                    icon={<DeleteIcon />}
+                    onClick={() => {
+                      openResetToken();
+                    }}
+                    color='red'
+                  >
+                    Reset Token
+                  </Menu.Item>
+                  <Menu.Item component={Link} icon={<LogoutIcon />} href='/auth/logout' color='red'>
+                    Logout
+                  </Menu.Item>
+                  <Menu.Divider />
+                  <>
+                    {oauth_providers
+                      .filter((x) =>
+                        user.oauth
+                          ?.map(({ provider }) => provider.toLowerCase())
+                          .includes(x.name.toLowerCase())
+                      )
+                      .map(({ name, Icon }, i) => (
+                        <>
+                          <Menu.Item
+                            closeMenuOnClick={false}
+                            key={i}
+                            icon={<Icon size={18} colorScheme={theme.colorScheme} />}
+                          >
+                            Logged in with {capitalize(name)}
+                          </Menu.Item>
+                        </>
+                      ))}
+                    {oauth_providers.filter((x) =>
+                      user.oauth?.map(({ provider }) => provider.toLowerCase()).includes(x.name.toLowerCase())
+                    ).length ? (
                       <Menu.Divider />
-                      <>
-                        {oauth_providers
-                          .filter((x) =>
-                            user.oauth
-                              ?.map(({ provider }) => provider.toLowerCase())
-                              .includes(x.name.toLowerCase())
-                          )
-                          .map(({ name, Icon }, i) => (
-                            <>
-                              <MenuItem
-                                sx={{ '&:hover': { backgroundColor: 'inherit' } }}
-                                key={i}
-                                py={5}
-                                px={4}
-                                icon={<Icon size={18} colorScheme={theme.colorScheme} />}
-                              >
-                                Logged in with {capitalize(name)}
-                              </MenuItem>
-                            </>
-                          ))}
-                        {oauth_providers.filter((x) =>
-                          user.oauth
-                            ?.map(({ provider }) => provider.toLowerCase())
-                            .includes(x.name.toLowerCase())
-                        ).length ? (
-                          <Menu.Divider />
-                        ) : null}
-                      </>
-                      <MenuItem icon={<PencilIcon />}>
-                        <Select
-                          size='xs'
-                          data={Object.keys(themes).map((t) => ({
-                            value: t,
-                            label: friendlyThemeName[t],
-                          }))}
-                          value={systemTheme}
-                          onChange={handleUpdateTheme}
-                        />
-                      </MenuItem>
-                    </Menu>
-                  </Stack>
-                </Popover.Dropdown>
-              </Popover>
+                    ) : null}
+                  </>
+                  <Menu.Item closeMenuOnClick={false} icon={<PencilIcon />}>
+                    <Select
+                      size='xs'
+                      data={Object.keys(themes).map((t) => ({
+                        value: t,
+                        label: friendlyThemeName[t],
+                      }))}
+                      value={systemTheme}
+                      onChange={handleUpdateTheme}
+                    />
+                  </Menu.Item>
+                </Menu.Dropdown>
+              </Menu>
             </Box>
           </div>
         </Header>

--- a/src/components/Layout.tsx
+++ b/src/components/Layout.tsx
@@ -396,75 +396,77 @@ export default function Layout({ children, props }) {
 
                 <Popover.Dropdown p={4} mr='md' sx={{ minWidth: '200px' }}>
                   <Stack spacing={2}>
-                    <Menu.Label>
-                      {user.username} ({user.id}){' '}
-                      {user.administrator && user.username !== 'administrator' ? '(Administrator)' : ''}
-                    </Menu.Label>
-                    <MenuItemLink icon={<SettingsIcon />} href='/dashboard/manage'>
-                      Manage Account
-                    </MenuItemLink>
-                    <MenuItem
-                      icon={<CopyIcon />}
-                      onClick={() => {
-                        setOpen(false);
-                        openCopyToken();
-                      }}
-                    >
-                      Copy Token
-                    </MenuItem>
-                    <MenuItem
-                      icon={<DeleteIcon />}
-                      onClick={() => {
-                        setOpen(false);
-                        openResetToken();
-                      }}
-                      color='red'
-                    >
-                      Reset Token
-                    </MenuItem>
-                    <MenuItemLink icon={<LogoutIcon />} href='/auth/logout' color='red'>
-                      Logout
-                    </MenuItemLink>
-                    <Menu.Divider />
-                    <>
-                      {oauth_providers
-                        .filter((x) =>
+                    <Menu>
+                      <Menu.Label>
+                        {user.username} ({user.id}){' '}
+                        {user.administrator && user.username !== 'administrator' ? '(Administrator)' : ''}
+                      </Menu.Label>
+                      <MenuItemLink icon={<SettingsIcon />} href='/dashboard/manage'>
+                        Manage Account
+                      </MenuItemLink>
+                      <MenuItem
+                        icon={<CopyIcon />}
+                        onClick={() => {
+                          setOpen(false);
+                          openCopyToken();
+                        }}
+                      >
+                        Copy Token
+                      </MenuItem>
+                      <MenuItem
+                        icon={<DeleteIcon />}
+                        onClick={() => {
+                          setOpen(false);
+                          openResetToken();
+                        }}
+                        color='red'
+                      >
+                        Reset Token
+                      </MenuItem>
+                      <MenuItemLink icon={<LogoutIcon />} href='/auth/logout' color='red'>
+                        Logout
+                      </MenuItemLink>
+                      <Menu.Divider />
+                      <>
+                        {oauth_providers
+                          .filter((x) =>
+                            user.oauth
+                              ?.map(({ provider }) => provider.toLowerCase())
+                              .includes(x.name.toLowerCase())
+                          )
+                          .map(({ name, Icon }, i) => (
+                            <>
+                              <MenuItem
+                                sx={{ '&:hover': { backgroundColor: 'inherit' } }}
+                                key={i}
+                                py={5}
+                                px={4}
+                                icon={<Icon size={18} colorScheme={theme.colorScheme} />}
+                              >
+                                Logged in with {capitalize(name)}
+                              </MenuItem>
+                            </>
+                          ))}
+                        {oauth_providers.filter((x) =>
                           user.oauth
                             ?.map(({ provider }) => provider.toLowerCase())
                             .includes(x.name.toLowerCase())
-                        )
-                        .map(({ name, Icon }, i) => (
-                          <>
-                            <MenuItem
-                              sx={{ '&:hover': { backgroundColor: 'inherit' } }}
-                              key={i}
-                              py={5}
-                              px={4}
-                              icon={<Icon size={18} colorScheme={theme.colorScheme} />}
-                            >
-                              Logged in with {capitalize(name)}
-                            </MenuItem>
-                          </>
-                        ))}
-                      {oauth_providers.filter((x) =>
-                        user.oauth
-                          ?.map(({ provider }) => provider.toLowerCase())
-                          .includes(x.name.toLowerCase())
-                      ).length ? (
-                        <Menu.Divider />
-                      ) : null}
-                    </>
-                    <MenuItem icon={<PencilIcon />}>
-                      <Select
-                        size='xs'
-                        data={Object.keys(themes).map((t) => ({
-                          value: t,
-                          label: friendlyThemeName[t],
-                        }))}
-                        value={systemTheme}
-                        onChange={handleUpdateTheme}
-                      />
-                    </MenuItem>
+                        ).length ? (
+                          <Menu.Divider />
+                        ) : null}
+                      </>
+                      <MenuItem icon={<PencilIcon />}>
+                        <Select
+                          size='xs'
+                          data={Object.keys(themes).map((t) => ({
+                            value: t,
+                            label: friendlyThemeName[t],
+                          }))}
+                          value={systemTheme}
+                          onChange={handleUpdateTheme}
+                        />
+                      </MenuItem>
+                    </Menu>
                   </Stack>
                 </Popover.Dropdown>
               </Popover>

